foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -110,65 +111,128 @@ public static OpenAiClient withCustomDestination(@Nonnull final Destination dest
     return client.withApiVersion(DEFAULT_API_VERSION);
   }
 
+  private static void throwOnContentFilter(@Nonnull final OpenAiChatCompletionDelta delta) {
+    final String finishReason = delta.getFinishReason();
+    if (finishReason != null && finishReason.equals("content_filter")) {
+      throw new OpenAiClientException("Content filter filtered the output.");
+    }
+  }
+
+  private static CreateChatCompletionRequest toCreateChatCompletionRequest(
```

Comment:
(Comment)

I appreciate this method remains private, in _orchestration_ its public unfortunately.